### PR TITLE
Order additional arguments after response file

### PR DESCRIPTION
**Pull request by chamons**
- https://github.com/xamarin/xamarin-macios/issues/4594
- There were being adding _before_ the response file
which meant options that were overriding defaults were
not being honored

### CODE REVIEW

File: tools/mmp/driver.cs [hunk 1]
```diff
@@ -499,6 +499,7 @@ static void Main2 (string [] args)
 			if (verbose > 0)
 				Console.WriteLine ("Selected target framework: {0}; API: {1}", targetFramework, IsClassic ? "Classic" : "Unified");
 
+			Log (1, $"Selected Linking: '{App.LinkMode}'");
```
Comment (chamons) [inline tools/mmp/driver.cs]: I'm technically just doing this so the test can easily tell, but it's really useful verbose info anyway, and we do it for other options as well.

**Comment by chamons** [comment]
I looked at writing an iOS test (it doesn't at first glance appear to be a bug there) but the current test infrastructure doesn't allow you to read stdout easily.

This is what I have so far. 

https://git.io/fNH12

If someone has a hint on how to continue I can keep hacking on it, or if someone in review insists that I need a test.

File: msbuild/Xamarin.Mac.Tasks.Core/Tasks/MmpTaskBase.cs [hunk 2]
```diff
@@ -237,9 +233,14 @@ protected override string GenerateCommandLineCommands ()
 				Log.LogWarning ("Failed to create response file '{0}': {1}", responseFile, ex);
 			}
 
-			// Use only the response file
+			// Reponse file must come before extra arguments
```
Comment (spouliot) [inline msbuild/Xamarin.Mac.Tasks.Core/Tasks/MmpTaskBase.cs]: minor: that's saying what's done (kind of obvious) but not why (which is less obvious)
don't update unless something else needs fixing (blame / git history will tell the story)

**Comment by chamons** [comment]
Bah, this is broken for iOS as well. I had XM installed from my dev build but iOS from stable (from working on another project). The stable version didn't use response files thus appeared to work.

```  Provided arguments:
        --linksdkonly
        -v
        -v
        @/Users/donblas/Programming/macios/master/xamarin-macios/msbuild/tests/AppWithExtraArgumentThatOverrides/obj/iPhoneSimulator/Debug/device-builds/iphone10.3-11.4/response-file.rsp
  ```

**Comment by spouliot** [comment]
@chamons the fix for XI can be in a separate PR

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs [hunk 3]
```diff
@@ -561,7 +562,7 @@ protected override string GenerateCommandLineCommands ()
 						}
 					} else {
 						// other user-defined mtouch arguments
-						actualArgs.AddQuoted (StringParserService.Parse (argument, customTags));
```
Comment (chamons) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]: I looked at refactoring all of this gcc flag bit out but it got large and messy quickly.

**Comment by spouliot** [comment]
@chamons please backport to 15.9 after merging. Thanks!

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs (hunk 3 above)
Comment (jstedfast) [reply msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]: s/seperatly/separately/g

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs [hunk 4]
```diff
@@ -618,9 +619,16 @@ protected override string GenerateCommandLineCommands ()
 				Log.LogWarning ("Failed to create response file '{0}': {1}", responseFile, ex);
 			}
 
-			// Use only the response file
+			// Some arguments can not safely go in the response file and are 
+			// added seperatly. They must go _after_ the response file
```
Comment (jstedfast) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]: s/seperatly/separately/

**Pull request merged**